File: app/src/main/java/com/cliniva/enventory/ui/more/MoreFragment.java
package com.cliniva.enventory.ui.more;


import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;

import com.cliniva.enventory.R;
import com.cliniva.enventory.ui.FragmentContainerActivity;
import com.cliniva.enventory.ui.base.BaseFragment;
import com.cliniva.enventory.utils.IntentUtils;

/**
 * A simple {@link Fragment} subclass.
 */
public class MoreFragment extends BaseFragment implements MoreContract.View, View.OnClickListener {

    private MoreContract.Presenter mMorePresenter;

    public MoreFragment() {
        // Required empty public constructor
    }

    public static MoreFragment getInstance(){
        MoreFragment fragment = new MoreFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.fragment_more;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mMorePresenter = new MorePresenter(this);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setToolbar(view, mMorePresenter.getContext().getString(R.string.action_more), false);

        ImageView btnEventTransaction = view.findViewById(R.id.btn_event_transaction);
        ImageView btnEventCustomer = view.findViewById(R.id.btn_event_customer);
        ImageView btnEventSupplier = view.findViewById(R.id.btn_event_supplier);

        btnEventTransaction.setOnClickListener(this);
        btnEventCustomer.setOnClickListener(this);
        btnEventSupplier.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.btn_event_transaction:
                onNavigation(0);
                break;
            case R.id.btn_event_customer:
                onNavigation(1);
                break;
            case R.id.btn_event_supplier:
                onNavigation(2);
                break;
        }
    }

    private void onNavigation(int extra){
        Bundle bundle = new Bundle();
        bundle.putInt("FRAGMENT_TYPE_EXTRA", extra);
        IntentUtils.getInstance().onActivityIntentWithExtras(mActivity, FragmentContainerActivity.class, bundle);
    }
}

File: app/src/main/java/com/amirsons/inventory/ui/fragment/more/MoreFragment.kt
package com.amirsons.inventory.ui.fragment.more

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.amirsons.inventory.R
import com.amirsons.inventory.ui.activity.FragmentContainerActivity
import com.amirsons.inventory.ui.activity.MainActivity
import com.amirsons.inventory.ui.activity.login.LoginActivity
import com.amirsons.inventory.ui.base.BaseFragment
import com.amirsons.inventory.utils.IntentUtils
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.fragment_more.*

/**
 * A simple [BaseFragment] subclass.
 */
class MoreFragment : BaseFragment(), MoreView, View.OnClickListener {

    private lateinit var mMorePresenter: MorePresenter

    override val contentLayout: Int
        get() = R.layout.fragment_more

    companion object {

        val instance: MoreFragment
            get() {
                val fragment = MoreFragment()
                val args = Bundle()
                fragment.arguments = args
                return fragment
            }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mMorePresenter = MoreMvp(this)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setToolbar(view, mMorePresenter.context.getString(R.string.action_more), false)

        btn_event_transaction.setOnClickListener(this)
        btn_event_customer.setOnClickListener(this)
        btn_event_supplier.setOnClickListener(this)
        tv_customers.setOnClickListener(this)
        tv_suppliers.setOnClickListener(this)

        btn_logout.setOnClickListener {

            FirebaseAuth.getInstance().signOut()

            val intent = Intent(context, LoginActivity::class.java)
            startActivity(intent)

            activity?.finish()
        }
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btn_event_transaction -> onNavigation(1)
            R.id.btn_event_customer, R.id.tv_customers -> onNavigation(2)
            R.id.btn_event_supplier, R.id.tv_suppliers -> onNavigation(3)
        }
    }

    private fun onNavigation(extra: Int) {
        val bundle = Bundle()
        bundle.putInt("FRAGMENT_TYPE_EXTRA", extra)
        IntentUtils.instance.onActivityIntentWithExtras(context, FragmentContainerActivity::class.java, bundle)
    }
}
